include/ceed.h

```diff
@@ -178,17 +178,17 @@ CEED_EXTERN int CeedErrorImpl(Ceed, const char *, int, const char *, int,
 
 /// Ceed error handlers
 CEED_EXTERN int CeedErrorReturn(Ceed, const char *, int, const char *, int,
-                                const char *, va_list);
+                                const char *, va_list *);
 CEED_EXTERN int CeedErrorStore(Ceed, const char *, int, const char *, int,
-                               const char *, va_list);
+                               const char *, va_list *);
 CEED_EXTERN int CeedErrorAbort(Ceed, const char *, int, const char *, int,
-                               const char *, va_list);
+                               const char *, va_list *);
 CEED_EXTERN int CeedErrorExit(Ceed, const char *, int, const char *, int,
-                              const char *, va_list);
+                              const char *, va_list *);
 CEED_EXTERN int CeedSetErrorHandler(Ceed ceed,
                                     int (*eh)(Ceed, const char *, int,
                                         const char *, int, const char *,
-                                        va_list));
+                                        va_list *));
```

Comment:
For the purposes of handling errors in Julia, I tentatively changed this interface to use `va_list *` instead of `va_list`. My understanding is that `va_list` is implementation defined, and so is hard to interface with using FFI (this will probably come up in the Rust interface, cc @jeremylt). This allows libCEED errors to be thrown and caught as Julia exceptions.

If people don't like this idea I'm happy to revert it.

(On a related note, the behavior of `CeedErrorStore` is a bit weird because it will be called first on `parent`, which will invalidate the `va_list`, so one of the `Ceed` objects will have a garbage message. This seems like a bug.)

Reply:
Did you see the latest fix in master where the error message is only set on the parent?

The message should be stored in the parent ceed and no other ceed. The user only has access to the parent ceed. and not the delegate ceeds created by the resource they specified, so those ceeds don't need to store the message.

Reply:
Good to know, thanks! I hadn't seen the fix. Just merged latest main into this branch.

Let me know what you think of the `va_list` pointer.

Reply:
We were mirroring `vprintf` conventions, but I'm not aware of any reason that passing a pointer would be problematic.

Reply:
It turns out that `clang-tidy` hates this

Reply:
Apparently there are [some bugs](https://bugs.llvm.org/show_bug.cgi?id=41311) in clang-tidy relating to `va_list`. I don't know if we're seeing the same thing. Could we use a typedef or something to make clang-tidy happy?

Reply:
Yeah, I saw those bugs. I'm personally fine with `// NOLINT`ing those lines since there is no critical logic in those 3 lines.